inspire/modules/citations/tasks.py

```diff
@@ -54,6 +55,41 @@ def update_records_citations(new_citations):
         citees.clear()
 
 
+@before_record_insert.connect
+def catch_citations_insert(sender, *args, **kwargs):
+    a = []
+    if 'references' in sender:
+        references = sender['references']
+        for reference in references:
+            recid = reference.get('recid')
+            if recid is not None:
+                a.append(recid)
+
+
+@before_record_update.connect
+def catch_citations_update(sender, *args, **kwargs):
+    a = []
+    if 'references' in sender:
+        references = sender['references']
+        for reference in references:
+            recid = reference.get('recid')
+            if recid is not None:
+                a.append(recid)
+    b = []
+    if 'control_number' in sender:
+        from invenio_ext.es import es
+        recid = sender['control_number']
+        rec = es.get(index='hep', id=recid)
```

Comment:
Indeed this could be a problem. Is there any chance that a 'hep' record and a record from another collection can never have the same id? If this is the case we can catch the 'NotFoundError' and return without doing anything. If not we have to think of something else.